commit.go

```diff
@@ -104,11 +104,27 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 	if err != nil {
 		return imgID, errors.Wrapf(err, "error computing layer digests and building metadata")
 	}
+
+	//TOM: Start
+	csrc, ok := src.(*containerImageRef) 
+	if ok {
+		fmt.Printf("TOM: src onbuild is %q\n", csrc.onBuild)
+	}
+	//TOM: end
+
 	// "Copy" our image to where it needs to be.
 	err = cp.Image(ctx, policyContext, dest, src, getCopyOptions(options.ReportWriter, nil, systemContext, ""))
 	if err != nil {
 		return imgID, errors.Wrapf(err, "error copying layers and metadata")
 	}
+	//TOM: Start
+	cdest, ok := dest.(*containerImageRef) 
+	if ok {
+		fmt.Printf("TOM: dest  onbuild is %q\n", cdest.onBuild)
+	}
+	dest.DockerReference().Config.OnBuild = csrc.onBuild
+	//TOM: end
+	fmt.Printf("TOM: AFTER dest onBuild )\n")
```

Comment:
This destination image never comes back OK.  I thought it would?  I was hoping I could push the OnBuild value into it here, but I'm not sure why I can't get to it.

Reply:
The destination isn't a `*containerImageRef`, since it's a reference to an image that's accessed via a transport that's supported by the image library, and image references are usually obtained by asking transport objects by way of `alltransports.ParseImageName()`, which uses the initial portion of the name to select which `ImageTransport` implementation to ask. (The `containers-storage` transport offers additional transport-specific methods for people who ask its `Transport` object directly.)

The `src` reference is a `*containerImageRef`, but it isn't obtained through a transport.  It's just enough of an `ImageReference` implementation that the image library can ask it for an `ImageSource`, which in turn is just enough of an `ImageSource` implementation that the image library can examine it and copy it.